src/blockchain/clients/node.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  AmountType,
+  Box,
+  SignedTransaction,
+  TokenId,
+  TransactionId,
+} from "@fleet-sdk/common";
+import { BlockchainClient } from "./blockchain_client.ts";
+import { Component } from "../../component.ts";
+import { ErgomaticConfig } from "../../config.ts";
+import axios, { AxiosInstance } from "axios";
+
+export class NodeClient extends Component implements BlockchainClient {
+  readonly #http: AxiosInstance;
+  #timeoutMs: number;
+
+  constructor(config: ErgomaticConfig, httpTimeoutMs: number = 10000) {
+    super(config, "NodeClient");
+
+    this.#timeoutMs = httpTimeoutMs;
+    this.#http = axios.create({
+      baseURL: config.node.endpoint,
+    });
+  }
+
+  getBlock(_height: number): Promise<unknown> {
+    throw new Error("Method not implemented.");
+  }
+
+  getCurrentHeight(): Promise<number> {
+    throw new Error("Method not implemented.");
+  }
+
+  submitTx(
+    signedTx: SignedTransaction,
+  ): Promise<TransactionId> {
+    return this.#http.post(
+      "/transactions",
+      signedTx,
+      this.#defaultRequestConfig,
+    );
+  }
+
+  // deno-lint-ignore require-yield
+  async *getBoxesByTokenId<T extends AmountType = string>(
+    _tokenId: TokenId,
+  ): AsyncGenerator<Box<T>[]> {
+    throw new Error(
+      `${this.name} does not support getBoxesByTokenId operation`,
+    );
+  }
+
+  getMempool(): Promise<SignedTransaction[]> {
+    // TODO: this might need pagination
```

Comment:
The endpoint `/transactions/unconfirmed/` doesn't impose a max amount for the `limit` param, we can take advantage of it and get the entire mempool in one request by passing a ridiculous big int there. Yeah, a bit hacky but performance matters :)

Reply:
done, I still implemented it as a paginated endpoint so if the limit is imposed we just need to decrease the limit query param - I imagine it may need to be imposed one day :smile: